common/setup.c

```diff
@@ -202,6 +201,19 @@ static void __text_init map_bios_area(void) {
         kmap_4k(bios_mfn, L1_PROT_RO);
 }
 
+static void display_cpu_info(void) {
+    char cpu_identifier[49];
+    unsigned long freq;
+
+    if (!cpu_vendor_string(cpu_identifier))
+        return;
+
+    printk("CPU: %.48s\n", cpu_identifier);
+    freq = get_cpu_freq(cpu_identifier);
+    if (freq > 0)
```

Comment:
What does it mean to get a freq = 0?
And should we treat that as error (getting freq == 0)?

Reply:
freq == 0 means we did not parse the frequency from processor brand string. This can happen when the brand string does not have frequency in it. It's not an error.